sycl/test/basic_tests/marray/marray.cpp

```diff
@@ -23,11 +23,45 @@ using namespace sycl;
   CHECK_ALIAS_BY_SIZE(ALIAS_MTYPE, ELEM_TYPE, 8)                               \
   CHECK_ALIAS_BY_SIZE(ALIAS_MTYPE, ELEM_TYPE, 16)
 
+#define CHECK_BINOP(OP, LHS, RHS)                                              \
+  assert((LHS[0] * RHS) == (LHS * RHS) && (LHS * RHS[0]) == (LHS * RHS));
```

Comment:
Do we need to add `LHS[0] * RHS[0] == LHS * RHS`? In either case, I think some comment would be helpful here to the reader.

Reply:
Added case and comment.